core/src/main/java/org/kohsuke/stapler/export/ExportConfig.java

```diff
@@ -1,41 +1,146 @@
 package org.kohsuke.stapler.export;
 
+import javax.annotation.Nonnull;
+
 /**
  * Controls the output behaviour.
  *
  * @author Kohsuke Kawaguchi
  */
 public class ExportConfig {
+    private final boolean prettyPrint;
+    private final ClassAttributeBehaviour classAttribute;
+    private final ExportInterceptor exportInterceptor;
+    private final boolean skipIfFail;
+    private final Flavor flavor;
+
     /**
-     * @deprecated
-     *      Use getter and setter
+     * Creates {@link ExportConfig} with a flavor
+     *
+     * @param flavor must be non-null
      */
-    public boolean prettyPrint;
-
-    private ClassAttributeBehaviour classAttribute = ClassAttributeBehaviour.IF_NEEDED;
-
-    public boolean isPrettyPrint() {
-        return prettyPrint;
+    private ExportConfig(Flavor flavor, ExportInterceptor exportInterceptor,
+                         ClassAttributeBehaviour classAttribute, boolean skipIfFail, boolean prettyPrint) {
+        this.flavor = flavor;
+        this.prettyPrint = prettyPrint;
+        this.classAttribute = (classAttribute == null) ? ClassAttributeBehaviour.IF_NEEDED.simple(): classAttribute;
+        this.exportInterceptor = exportInterceptor == null ? ExportInterceptor.DEFAULT :exportInterceptor;
+        this.skipIfFail = skipIfFail;
     }
 
     /**
      * If true, output will be indented to make it easier for humans to understand.
      */
-    public ExportConfig withPrettyPrint(boolean prettyPrint) {
-        this.prettyPrint = prettyPrint;
-        return this;
+    public boolean isPrettyPrint() {
+        return prettyPrint;
     }
 
+    /**
+     * Controls the behaviour of the class attribute to be produced.
+     */
     public ClassAttributeBehaviour getClassAttribute() {
         return classAttribute;
     }
 
     /**
      * Controls the behaviour of the class attribute to be produced.
      */
-    public ExportConfig withClassAttribute(ClassAttributeBehaviour cab) {
```

Comment:
@oleg-nenashev ok, yeah its exposed in `Flavor.createDataWriter()` public methods and can be used from outside.  I missed that, will revert it as I don't want to conflate this PR and keep it focused on what its trying to solve. cc: @jglick.